File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03017/p03017_num3_parsed.cpp
#include <iostream>
#include <string>

int main() {
    int N, A, B, C, D;
    std::cin >> N >> A >> B >> C >> D;
    std::string S;
    std::cin >> S;

    // Check if there is any rock between B and C
    for (int i = B - 1; i < C - 1; i++) {
        if (S[i] == '#' && S[i + 1] == '#') {
            std::cout << "No" << std::endl;
            return 0;
        }
    }

    // Check if Snuke can reach C before Fnuke reaches D
    if (C < D) {
        for (int i = B - 1; i < D - 1; i++) {
            if (S[i] == '.' && S[i + 1] == '.' && S[i + 2] == '.') {
                std::cout << "Yes" << std::endl;
                return 0;
            }
        }
        std::cout << "No" << std::endl;
        return 0;
    }

    // Check if Fnuke can reach D before Snuke reaches C
    if (C > D) {
        for (int i = B; i < C - 1; i++) {
            if (S[i] == '.' && S[i + 1] == '.' && S[i + 2] == '.') {
                std::cout << "Yes" << std::endl;
                return 0;
            }
        }
        std::cout << "No" << std::endl;
        return 0;
    }

    std::cout << "Yes" << std::endl;
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03017/p03017_num3_parsed.py
# Program

def can_jump(N, A, B, C, D, S):
    # Check if there are any rocks between A and C
    for i in range(A, C):
        if S[i-1] == '#':
            return "No"
    
    # Check if there are any rocks between B and D
    for i in range(B, D):
        if S[i-1] == '#':
            return "No"
    
    # Check if Snuke is always ahead of Fnuke
    if B < C and D < C:
        return "Yes"
    else:
        return "No"

# Read input
N, A, B, C, D = map(int, input().split())
S = input()

# Call the function and print the result
print(can_jump(N, A, B, C, D, S))